### PR TITLE
Convert console.error writes to exceptions

### PR DESCRIPTION
**Pull request by ericpromislow**
* Verify minimum and specified values are sane wrt max

* Redefine console.error to spy on and verify errors.

Github wouldn't let me reopen PR #85 due to a force-push (?). This covers the same issue.

### CODE REVIEW

**Comment by ericpromislow** [comment]
Note: this PR also contains changes that got into main earlier but were then overwritten, probably during rebasing

File: src/pages/K8s.vue [hunk 1]
```diff
@@ -158,7 +155,11 @@ export default {
       that.$data.state = stt;
     });
     ipcRenderer.on('settings-update', (event, settings) => {
-      this.$data.settings = settings;
+      // TODO: put in a status bar
+      console.log('settings have been updated');
+
+      // NOOOOO:!
```
Comment (mook-as) [inline src/pages/K8s.vue:161]: Please elaborate; _why_ is this a no?

**Review by mook-as — CHANGES_REQUESTED** [review]

File: src/components/__tests__/SystemPreferences.spec.js [hunk 2]
```diff
@@ -89,4 +90,97 @@ describe('SystemPreferences.vue', () => {
     expect(slider2.exists()).toBeTruthy();
     expect(slider2.find('div.vue-slider-rail div.vue-slider-dot.vue-slider-dot-disabled').exists()).toBeTruthy();
   });
+
+  describe('throw on console.error', () => {
+    class VueError extends Error {
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/components/__tests__/SystemPreferences.spec.js:95]: In the sense that we're testing for the vue-slider-component to write to console.error when something goes wrong, and it tags all its lines with `[VueSlider Error]`, there is.  `VueSliderError` would be a better name.

File: src/components/__tests__/SystemPreferences.spec.js [hunk 3]
```diff
@@ -89,4 +90,97 @@ describe('SystemPreferences.vue', () => {
...
+        throw new VueError(...args);
+      };
+    });
+    afterAll(() => {
+      console.error = origError;
+    });
+
+    const checkForError = (func, expectedMessage) => {
+      expect(func).toThrowError(VueError);
+      expect(func).toThrowError(expectedMessage);
```
[the remark this answers is not in the document]
Comment (ericpromislow) [reply src/components/__tests__/SystemPreferences.spec.js:111]: I didn't see that way of doing a single test. Thanks.

As for mocks and spies, I went with the simplest possible way that would work. Testing with the throws meant I didn't need to use any auxiliary variables or calls to `mockClear`

File: src/pages/K8s.vue (hunk 1 above)
Comment (ericpromislow) [reply src/pages/K8s.vue:161]: This was my knee-jerk response from when the sliders were free-form fields, but I'd like to keep the model like so:
* when a form is shown, its fields are initialized based on prefs and current state
* field values change only by user action
* if saving a field requires making some change when saving it, but there's no need to change how it's stored in the field, don't change it under the user's feet.

I'll re-evaluate that with the current set of widgets.

Comment (mook-as) [reply src/pages/K8s.vue:161]: I'd be happy with a similarly verbose comment in the code.

**Review by mook-as — APPROVED** [review]

**Review by mook-as — CHANGES_REQUESTED** [review]
Oops missed the Noooo

File: src/pages/K8s.vue (hunk 1 above)
Comment (ericpromislow) [reply src/pages/K8s.vue]: Turns out we can currently update the prefs on update with no problems. This was the behavior when the only setting changed the version field. So let's leave it that way until something breaks.

**Review by mook-as — APPROVED** [review]

**Pull request merged**